docs/mdbook/default.nix

```
@@ -67,12 +67,13 @@ let
           filterAttrs (_: component: component.isOption && (isVisible component)) opts;
       path = removeWhitespace (concatStringsSep "/" path);
       moduleDoc =
-        if builtins.length path >= 2 && lib.hasAttrByPath path nixvimInfo then
-          let
-            info = lib.getAttrFromPath path nixvimInfo;
-            maintainers = lib.unique (options.config.meta.maintainers.${info.file} or [ ]);
-            maintainersNames = builtins.map (m: m.name) maintainers;
-          in
+        let
+          info = optionalAttrs (hasAttrByPath path nixvimInfo) (getAttrFromPath path nixvimInfo);
+          maintainers = lib.unique (options.config.meta.maintainers.${info.file} or [ ]);
+          maintainersNames = builtins.map (m: m.name) maintainers;
+        in
+        # Validate this actually is an info attrset
+        if info ? file && info ? description && info ? url then
```

Alternative: slightly more explicit:

```suggestion
        if attrNames info == [ "file" "description" "url" ] then
```

---

Yeah I agree that it would be more readable. The other way would be to use a `__nixvim` option that would mark this, but it is uglier

---

On second thoughts, nixfmt breaks this over several lines **and** the list order is actually semantically significant, which could easily turn into a bug one day...

```nix
# nixfmt:
if
  attrNames info == [
    "description"
    "file"
    "url"
  ]
then
```

I think I'll leave it as it was after all.